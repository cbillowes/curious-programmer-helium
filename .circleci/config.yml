version: 2
jobs:
  build:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Chopping onions without the tears"
      - run: npm run build
  test:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making soup and drinking beers"
      - run: sudo npm install -g jest-cli
      - run: npm run test
  page_speed:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making sure it won't take years"
      - run: echo "* Deploy to Netlify"
      - run: echo "* Execute lighthouse page speed tests on build"
workflows:
  version: 2
  integration:
    jobs:
      - build
      - test:
          requires:
            - build
      - page_speed:
          requires:
            - test
