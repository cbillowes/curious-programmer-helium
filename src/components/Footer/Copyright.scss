.footer-container {
    .footer {
        .copyright-footer {
            font-size: 0.8em;
        }
    }
}
