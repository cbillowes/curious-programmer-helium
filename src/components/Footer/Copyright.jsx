import React, { Component } from "react"
import Link from "gatsby-link"
import config from "../../../data/SiteConfig"
import "./Copyright.scss"

class License extends Component {
  render() {
    return (
      <div className="copyright-footer">
        Copyright &copy; { "" } 
        {new Date().getFullYear()} { "" }
        {config.siteTitle}. { "" }
        Made with Gatsby v2 and other { "" }
        <Link to="/credits">cool stuff</Link>.
      </div>
    )
  }
}

export default License
