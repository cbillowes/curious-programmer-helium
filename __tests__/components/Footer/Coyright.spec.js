import $ from "jquery"
import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Copyright from "../../../src/components/Footer/Copyright"
import config from "../../../data/SiteConfig"

describe("a copyright component", () => {
  let element = <Copyright />
  let component
  
  beforeAll(() => {
    component = mount(element)
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render container", () => {
    expect(component.find(".copyright-footer").length).toEqual(1)
  })

  it("should render copyright content", () => {
    let footerText = `Made with Gatsby v2 and other <a href="/credits">cool stuff</a>.`
    let expected = `Copyright Â© ${new Date().getFullYear()} ${config.siteTitle}. ${footerText}`
    expect(component.html()).toContain(expected)
  })
})