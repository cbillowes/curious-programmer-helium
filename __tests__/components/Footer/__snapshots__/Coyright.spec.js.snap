// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a copyright component should render 1`] = `
<div
  className="copyright-footer"
>
  Copyright Â© 
  
  2019
   
  
  Curious Programmer
  . 
  
  Made with Gatsby v2 and other 
  
  <a
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    cool stuff
  </a>
  .
</div>
`;
